R/aggregate_alignment_loanbook_exposure.R

```diff
@@ -5,46 +5,127 @@
 #' @param level Character. Vector that indicates if the aggregate alignment
 #'   metric should be returned based on the net technology deviations (`net`) or
 #'   disaggregated into buildout and phaseout technologies (`bo_po`).
+#' @param .by <tidy-select> Character. Optionally, a selection of columns to
+#'   group by, instead. All columns indicated must be available variables in the
+#'   `matched` data set. The intended use case is to allow analyzing the loan
+#'   books by additional traits of interest, such as types of financial
+#'   institutions.
 #'
 #' @return NULL
 #' @export
 aggregate_alignment_loanbook_exposure <- function(data,
                                                   matched,
-                                                  level = c("net", "bo_po")) {
-  group_vars <- c("group_id", "scenario", "region", "sector", "year", "direction")
+                                                  level = c("net", "bo_po"),
+                                                  .by = NULL) {
+  group_vars <- c(
+    "scenario",
+    "region",
+    "sector",
+    "year",
+    "direction"
+  )
   level <- rlang::arg_match(level)
 
   # validate input data sets
   validate_input_data_aggregate_alignment_loanbook_exposure(
     data = data,
     matched = matched,
-    group_vars = group_vars
+    group_vars = group_vars,
+    .by = .by
   )
 
+  if (!is.null(.by)) {
+    if (!inherits(.by, "character")) {
+      stop(glue::glue("`.by` must be a character vector. Your input is {class(.by)}."))
+    }
+    group_vars <- c(.by, group_vars)
+  }
```

Comment:
NIT: Does this work if you pass `.by` a `tidyselect` function? (e.g. `.by = starts_with("x")`)? I genuinely don't know haha

Reply:
turns out that it does not.. I guess I will need to adapt the documentation